File: Dart/URL/get_url_scheme.dart
import "dart:core";

void main() {
  // Parse a url scheme.
  print(getURLScheme("http://www.example.com"));
}

// Parse a url scheme from a given url and return it.
String getURLScheme(final String url) {
  final Uri parsedURL = Uri.parse(url);
  return parsedURL.scheme;
}

File: Go/URL/get-url-scheme.go
package main

import (
	"fmt"
	"log"
	"net/url"
)

func main() {
	// Get the scheme of a url.
	fmt.Println(getURLScheme("https://example.com:8080"))
}

// Get the scheme of a given url.
func getURLScheme(givenURL string) string {
	url, err := url.Parse(givenURL)
	if err != nil {
		log.Fatalln(err)
	}
	return url.Scheme
}
